pages/api/mail/index.ts

```diff
@@ -0,0 +1,111 @@
+import { ResponseUtil } from "@utils/responseUtil";
+import type { NextApiRequest, NextApiResponse } from "next";
+import send from "services/nodemailer/mail";
+import findEmails from "@database/mail";
+
+/**
+ * Handles the request to /api/mail (POST)
+ * @param req request object
+ * @param res response object
+ * @returns Promise<void> and sends json object back to the client (type: ResponseObject)
+ */
+export default async function mailHandler(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    if (req.method == "POST" && req.body.key == process.env.AWS_LAMBDA_KEY) {
+        // AWS_LAMBDA_SECRET
+        const threeHours = 3;
+        const fortyEightHours = 48;
+        // finds all classes starting in 2-4 and 47-49 hours from now
+        // as well as their associated parents/volunteers information
+        const classesInThreeHours = await findEmails(threeHours);
+        const classesInFortyEightHours = await findEmails(fortyEightHours);
+        // stores all promises for the nodemailer transport
+        const mailerPromises = [];
+        // looping through all the classes starting in three hours
+        for (let i = 0; i < classesInThreeHours.length; ++i) {
+            // looping through parents that are registered to EACH class
+            for (let j = 0; j < classesInThreeHours[i].parentRegs.length; ++j) {
+                mailerPromises.push(
+                    send(
+                        process.env.EMAIL_FROM,
+                        classesInThreeHours[i].parentRegs[j].parent.user.email,
+                        "Reminder: Social Diversity for Children Class In 3 Hours",
+                        `<p>Hi ${classesInThreeHours[i].parentRegs[j].parent.user.firstName},</p>
+                        <p>The class <b>${classesInThreeHours[i].name}</b> you signed up for is starting in 3 hours!</p>
+                        <p>Regards, Social Diversity for Children</p>`,
```

Comment:
nit: construct this as a variable rather than passing it in as a big string literal